lib/online_order.rb

```diff
@@ -0,0 +1,104 @@
+require 'csv'
+require 'awesome_print'
+require 'money'
+require_relative '../lib/order'
+require_relative '../lib/customer'
+I18n.enforce_available_locales = false
+
+module Grocery
+  class OnlineOrder < Order
+    attr_reader :id, :products, :customer, :status
+
+    def initialize(id, products, customer, status)
+      super(id, products)
+      @customer = customer
+      @status = status
+    end
+
+    def total
+      sum_with_tax_fee = super
+      if sum_with_tax_fee != Money.new(0, "USD")
+        sum_with_tax_fee += Money.new(1000, "USD")
+      end
+      return sum_with_tax_fee
+    end
+
+    def add_product(product_name, product_price)
+      if @status == :pending || @status == :paid
+        true_false = super(product_name, product_price)
+        # true if product added, false if product not added
+        return true_false
+      end
+      # unsure how to make this an argument error instead of a standard error
+      raise StandardError.new("Error: Can't add product. Order status is #{@status}.")
+    rescue StandardError => e
+      return e.message
+    end
+
+    def self.all
+      online_orders_entered = []
+      CSV.read("/Users/brandyaustin/ada/week2/grocery_store/grocery-store/support/online_orders.csv").each do |row|
+        id = row[0].to_i
+        products = {}
+        products_array = row[1].split(";")
+
+        products_array.each do |item|
+          name, price = item.split(':')
+          products[name] = price.to_f
+        end
+
+        customer = "placeholder"
+        Grocery::Customer.all.each do |customer_object|
+          if customer_object.id == row[2].to_i
+            customer = customer_object
+          end
+        end
+
+        if row[3].nil?
+          status = :pending
+        else
+          status = row[3].to_sym
+        end
+        online_orders_entered << OnlineOrder.new(id, products, customer, status)
+      end
+      return online_orders_entered
+    end
+
+    def self.find(id)
```

Comment:
Notice that this method should be **exactly** the same as the one in `Order`... You can just use the superclass' version and **not** override it.